From 4c79be944b98c6775f224e013a41d0960b65ee5a Mon Sep 17 00:00:00 2001
From: Michal Hrusecky <michal.hrusecky@turris.com>
Date: Fri, 5 Nov 2021 17:02:54 +0100
Subject: [PATCH] init: Behave correctly when encountering error

Fix the condition so people with problems wouldn't end up with tmpfs on
top of btrfs.
---
 scripts/srv.init | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/srv.init b/scripts/srv.init
index 329eb6b..4d6e868 100644
--- a/scripts/srv.init
+++ b/scripts/srv.init
@@ -9,7 +9,7 @@ status() {
 
 die() {
     create_notification -n error "$1"
-    if grep -q '^[^[:blank:]]\+[[:blank:]]\+/srv[[:blank:]]' /proc/mounts && [ -d /srv ]; then
+    if ! grep -q '^[^[:blank:]]\+[[:blank:]]\+/srv[[:blank:]]' /proc/mounts && [ -d /srv ]; then
        mount -t tmpfs -o size=50M none /srv
     fi
     exit 1
-- 
GitLab

